hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -547,6 +547,11 @@ protected Map<String, String> properties() {
     return catalogProperties == null ? ImmutableMap.of() : catalogProperties;
   }
 
+  public void close() {
+    fileIO.close();
+    clients.close();
```

Comment:
The current practice is that we use `CatalogUtil.loadCatalog` to create the Catalog every time when we load a table. Opening and closing the clientPool (HMSClients) every time would be a costly operation, and we would likely lose everything that we win by using a pool.

Basically the expectation is that we have a global catalog cache and a thin Catalog wrapper using it.

Reply:
This is why I did not link this method into existing code paths and left it completely optional.
For users of `HiveCatalog` there needs to be a way to close the resources.

Reply:
So when I have a `HiveCatalog`, I can just throw it away? This seems strange with resources inside it that seemingly need to have `close`.

Reply:
Took some time to wrap my head around this, but that's the current situation